src/pytest_memray/marks.py

```diff
@@ -271,7 +272,7 @@ def limit_leaks(
 class _TrackedObjectsInfo:
     """Type that holds information about objects that survived tracking."""
 
-    surviving_objects: list
+    surviving_objects: list[Any]
```

Comment:
Shouldn't it be `list[object]`, not `list[Any]`? Likewise for every occurrence of `Any` that you've added.